### PR TITLE
docs: Polish language specification

### PR DESCRIPTION
**Pull request by juliaogris**
- Title-case all titles in spec
- Change italics to bold in spec and builtins

Add some typo fixes and minor improvements on various spec sections.

---

Review link: https://github.com/foxygoat/evy/blob/polish-spec/docs/spec.md

### CODE REVIEW

File: docs/spec.md [hunk 1]
```diff
@@ -272,9 +272,11 @@ according to the IEEE-754 64-bit floating point standard.
 
 ## Variables and Declarations
 
-Variables hold values of a specific type. They must be _declared_ before
-they can be used. There are two types of variable declarations:
-inferred declarations and typed declarations.
+Variables hold values of a specific type. They must be _declared_
+before they can be used. A declared variable mus be used at least once,
```
Comment (camh-) [inline docs/spec.md:276]: s/mus/must/

File: docs/spec.md [hunk 2]
```diff
@@ -1302,8 +1302,9 @@ y []any
 
 ## Type Assertion
 
-A type assertion `ident.(type)` asserts that the value of the variable
-`ident` is of the given `type`. If the assertion does not hold a
+A type assertion `x.(TYPE)` asserts that the value of the variable `x`
+is of the given `TYPE`. `TYPE` can be any basic or composite type, such
+as `num` or `[]string`. If the assertion does not hold a
```
Comment (camh-) [inline docs/spec.md:1307]: I think I would put a comma after "hold" to disambiguate "does not hold" and "does not hold a <noun>"

**Review by camh- — APPROVED** [review]
:four_leaf_clover: LGTM with a couple of minor comments

File: docs/spec.md (hunk 2 above)
Comment (juliaogris) [reply docs/spec.md:1307]: agreed, done.

File: docs/spec.md (hunk 1 above)
Comment (juliaogris) [reply docs/spec.md:276]: done, thanks.

**Pull request merged**